File: Person-Machine Interfaces/P2/app/lib/models/posology.dart
class Posology {
  final int? id;
  final int hour;
  final int minute;
  final int medicationId;

  // Constructor
  Posology({
    this.id,
    required this.hour,
    required this.minute,
    required this.medicationId,
  });

  // JSON to object
  factory Posology.fromJson(Map<String, dynamic> json) {
    return Posology(
      id: json['id'],
      hour: json['hour'],
      minute: json['minute'],
      medicationId: json['medication_id'],
    );
  }

  // Object to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'hour': hour,
      'minute': minute,
      'medication_id': medicationId,
    };
  }

  @override
  String toString() {
    return 'Posology(id: $id, hour: $hour, minute: $minute, medicationId: $medicationId)';
  }
}

File: IPM/P1/src/entities/posology.py
from typing import Optional

from gi.repository import GObject


class Posology(GObject.GObject):
    def __init__(self, id: Optional[int], hour: int, minute: int, medication_id: int) -> None:
        super().__init__()
        self.id = id
        self.hour = hour
        self.minute = minute
        self.medication_id = medication_id